test/rtp_decoder.c

```diff
@@ -671,32 +702,51 @@ void rtp_decoder_handle_pkt(u_char *arg,
     }
     const void *rtp_packet = bytes + dcdr->rtp_offset;
 
-    memcpy((void *)&dcdr->message, rtp_packet, hdr->caplen - dcdr->rtp_offset);
+    memcpy((void *)&message, rtp_packet, hdr->caplen - dcdr->rtp_offset);
     pktsize = hdr->caplen - dcdr->rtp_offset;
     octets_recvd = pktsize;
 
     if (octets_recvd == -1) {
         return;
     }
 
-    /* verify rtp header */
-    if (dcdr->message.header.version != 2) {
-        return;
+    if (dcdr->mode == mode_rtp) {
+        rtp = 1;
+    } else if (dcdr->mode == mode_rtcp) {
+        rtp = 0;
+    } else {
+        rtp = 1;
+        if (octets_recvd >= 2) {
+            /* rfc5761 */
+            u_char pt = *(bytes + dcdr->rtp_offset + 1) & 0x7f;
+            rtp = 64 <= pt && pt >= 95;
```

Comment:
Is not the `64 <= pt` check redundant here since the expression `pt >= 95` will be `false` for all values from 0 to 94 (assuming the range 0 -> 127 due to the `0x7f` mask).

From the RFC:

   Given these constraints, it is RECOMMENDED to follow the guidelines
   in the RTP/AVP profile [7] for the choice of RTP payload type values,
   with the additional restriction that payload type values in the range
   64-95 MUST NOT be used.

Seems like the check should allow anything outside the range 64-95 if I'm looking at the right part of the RFC.

`rtp = pt < 64 || pt > 95;`? Not sure, but something seems a bit off here.

Reply:
I think to check for RTCP, it should be `pt >= 64 && pt <= 95`.

That said, if we're muxing RTCP with RTP, there might also be DTLS or STUN packets. Should we also check the first octet in the packet?  In addition to the PT value, the first octet of the packet should be `octet1 >= 128 && octet1 <= 191`.  If that first octet isn't in that range, it's neither RTP nor RTCP, so the packet should be skipped over entirely.

Reply:
> If that first octet isn't in that range, it's neither RTP nor RTCP, so the packet should be skipped over entirely.

That makes sense to me, the packet captures I have been testing with have STUN packets along with the RTP+RTCP MUX.

Reply:
@thisisG , you are right I will replace it with pt < 64 || pt > 95 .
@paulej , any packet that is not SRTP or SRTCP will simply fail the srtp_unprotect functions provided authentication is enabled. The purpose of this check was to ensure the correct srtp_unprotect function was called. Providing any other RTP or RTCP validation is possible but maybe outside the scope of this change?

Reply:
@kyzer-davis, maybe you would like to confirm it still works for you after latest change?

Reply:
Sure:
Tested a decode of SRTCP+SRTP MUX
Tested a decode of SRTP stream with a few different payload types on the same UDP source port. G711 RTP (0), RTP-NTE DTMF (101), Comfort Noise (13).

They both decoded without errors as per my manual checks.
The new rtp, rtcp, error counters also appear to be in line with my manual checks.